<a href="https://colab.research.google.com/github/AnnaVithoriaMS/Enegia_renovavel/blob/main/Projeto_EnergiaRenovavel_ADA.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [5]:
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import numpy as np
import warnings
import plotly.express as px
import plotly.io as pio

In [6]:
warnings.filterwarnings('ignore')

 # Produção de Energias Renováveis


Energias renováveis são fontes naturais de energia capaz
  de se renovar, diferentemente de  energias de fonte não renovavel, como os combustíveis fósseis.  Exemplos de energia renovável incluem solar, eólica, hidrelétrica, geotérmica e biomassa.
    Assim as fontes são abundantes e inesgotáveis, alem de serem limpas, já que produzem menos ou nenhum gás de efeito estufa, no entanto é importante resaltar que a produção é afeta por codições naturais, alem de dificuldades para armazenamento e distribuição.

In [7]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [8]:
df_ener = pd.read_csv('/content/drive/MyDrive/producao_energia_mundo.csv')

In [9]:
df_ener

Pais Code   Ano  Eolica  Hidroeletrica  Solar  Outras Fontes  \
0     Afghanistan  AFG  2000     0.0           0.31   0.00           0.00   
1     Afghanistan  AFG  2001     0.0           0.50   0.00           0.00   
2     Afghanistan  AFG  2002     0.0           0.56   0.00           0.00   
3     Afghanistan  AFG  2003     0.0           0.63   0.00           0.00   
4     Afghanistan  AFG  2004     0.0           0.56   0.00           0.00   
...           ...  ...   ...     ...            ...    ...            ...   
6935     Zimbabwe  ZWE  2016     0.0           2.95   0.01           0.36   
6936     Zimbabwe  ZWE  2017     0.0           3.93   0.01           0.32   
6937     Zimbabwe  ZWE  2018     0.0           5.00   0.01           0.39   
6938     Zimbabwe  ZWE  2019     0.0           7.26   0.01           0.38   
6939     Zimbabwe  ZWE  2020     0.0           7.26   0.01           0.38   

      region          sub-region  
0       Asia       Southern Asia  
1       Asia       Southern Asia  
2       Asia       Southern Asia  
3       Asia       Southern Asia  
4       Asia       Southern Asia  
...      ...                 ...  
6935  Africa  Sub-Saharan Africa  
6936  Africa  Sub-Saharan Africa  
6937  Africa  Sub-Saharan Africa  
6938  Africa  Sub-Saharan Africa  
6939  Africa  Sub-Saharan Africa  

[6940 rows x 9 columns]

In [10]:
df_ener.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6940 entries, 0 to 6939
Data columns (total 9 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Pais           6940 non-null   object 
 1   Code           6940 non-null   object 
 2   Ano            6940 non-null   int64  
 3   Eolica         6940 non-null   float64
 4   Hidroeletrica  6940 non-null   float64
 5   Solar          6940 non-null   float64
 6   Outras Fontes  6936 non-null   float64
 7   region         6940 non-null   object 
 8   sub-region     6940 non-null   object 
dtypes: float64(4), int64(1), object(4)
memory usage: 488.1+ KB


In [11]:
df_ener=df_ener.fillna(0)

In [12]:
df_ener['Total_Energia_Renovavel'] = df_ener[['Eolica','Hidroeletrica','Solar','Outras Fontes']].sum(axis=1)

In [13]:
df_ener

Pais Code   Ano  Eolica  Hidroeletrica  Solar  Outras Fontes  \
0     Afghanistan  AFG  2000     0.0           0.31   0.00           0.00   
1     Afghanistan  AFG  2001     0.0           0.50   0.00           0.00   
2     Afghanistan  AFG  2002     0.0           0.56   0.00           0.00   
3     Afghanistan  AFG  2003     0.0           0.63   0.00           0.00   
4     Afghanistan  AFG  2004     0.0           0.56   0.00           0.00   
...           ...  ...   ...     ...            ...    ...            ...   
6935     Zimbabwe  ZWE  2016     0.0           2.95   0.01           0.36   
6936     Zimbabwe  ZWE  2017     0.0           3.93   0.01           0.32   
6937     Zimbabwe  ZWE  2018     0.0           5.00   0.01           0.39   
6938     Zimbabwe  ZWE  2019     0.0           7.26   0.01           0.38   
6939     Zimbabwe  ZWE  2020     0.0           7.26   0.01           0.38   

      region          sub-region  Total_Energia_Renovavel  
0       Asia       Southern Asia                     0.31  
1       Asia       Southern Asia                     0.50  
2       Asia       Southern Asia                     0.56  
3       Asia       Southern Asia                     0.63  
4       Asia       Southern Asia                     0.56  
...      ...                 ...                      ...  
6935  Africa  Sub-Saharan Africa                     3.32  
6936  Africa  Sub-Saharan Africa                     4.26  
6937  Africa  Sub-Saharan Africa                     5.40  
6938  Africa  Sub-Saharan Africa                     7.65  
6939  Africa  Sub-Saharan Africa                     7.65  

[6940 rows x 10 columns]

In [14]:
def mapa_energia(ener):
    aggs = ["sum","avg","median","mode","min","max"]

    agg = []
    agg_func = []
    for i in range(0, len(aggs)):
        agg = dict(
            args=['transforms[0].aggregations[0].func', aggs[i]],
            label=aggs[i],
            method='restyle'
        )
        agg_func.append(agg)

    data = [dict(
      type = 'choropleth',
      locationmode = 'country names',
      locations = df_ener['Pais'],
      z = df_ener[ener],
      autocolorscale = False,
      colorscale = 'Portland',
      reversescale = True,
      transforms = [dict(
        type = 'aggregate',
        groups = df_ener['Pais'],
        aggregations = [dict(
            target = 'z', func = 'sum', enabled = True)
        ]
      )]
    )]

    layout = dict(
      title = f'Energia {ener} nos países',
      xaxis = dict(title = 'Subject'),
      yaxis = dict(title = 'Score', range = [0,22]),
      height = 600,
      width = 900,
      updatemenus = [dict(
            x = 0.85,
            y = 1.15,
            xref = 'paper',
            yref = 'paper',
            yanchor = 'top',
            active = 1,
            showactive = False,
            buttons = agg_func
      )]
    )

    fig_dict = dict(data=data, layout=layout)

    pio.show(fig_dict, validate=False)


In [15]:
mapa_energia('Eolica')

In [16]:
mapa_energia('Hidroeletrica')

In [17]:
mapa_energia('Solar')

In [18]:
mapa_energia('Outras Fontes')

In [19]:
mapa_energia('Total_Energia_Renovavel')

## Africa
Para além da energia solar, os recursos renováveis de África incluem a energia eólica, a biomassa, a hidroeletricidade e minerais como lítio, grafite e cobalto, que são necessários para as tecnologias de energias renováveis, incluindo a produção de painéis solares e baterias para veículos elétricos

Fonte: <https://www.afdb.org/pt>

In [20]:
df_af = df_ener[df_ener['region'] == 'Africa']


In [21]:
fig = px.histogram(df_af, x="Ano", y="Total_Energia_Renovavel",text_auto='.2s',

                   title = 'Soma do total da produção de energia renovavel em TW por Ano no países da África',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

In [22]:
df_af = df_af[df_af['Ano'] > 1999]

In [23]:
fig = px.line(df_af, x="Ano", y="Total_Energia_Renovavel", color='Pais')
fig.show()

In [24]:
df_eu = df_ener[df_ener['region'] == 'Europe']

In [25]:
fig = px.histogram(df_eu, x="Ano", y="Total_Energia_Renovavel",text_auto='.2s',

                   title = 'Soma do total da produção de energia renovavel em TW por Ano em Paises da Europa',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

In [26]:
fig = px.line(df_eu, x="Ano", y="Total_Energia_Renovavel", color='Pais')
fig.show()

## China


In [27]:
df_ch = df_ener[df_ener['Pais'] == 'China']

In [28]:
data_CH = pd.melt(df_ch, id_vars=['Ano'], value_vars=['Eolica','Hidroeletrica','Solar','Outras Fontes','Total_Energia_Renovavel'], var_name='Fontes', value_name='Producao')

In [29]:
fig = px.line(data_CH, x="Ano", y="Producao", color='Fontes')
fig.show()

## EUA

In [30]:
df_eu = df_ener[df_ener['Code'] == 'USA']

In [31]:
data_EU = pd.melt(df_eu, id_vars=['Ano'], value_vars=['Eolica','Hidroeletrica','Solar','Outras Fontes','Total_Energia_Renovavel'], var_name='Fontes', value_name='Producao')

In [32]:
df = pd.read_csv('/content/drive/MyDrive/producao_energia_America_do-Sul.csv')

In [33]:
df

Pais Code   Ano  Eolica  Hidroeletrica  Solar  Outras Fontes  \
0    Argentina  ARG  2000    0.04          28.49   0.00           0.70   
1    Argentina  ARG  2001    0.05          36.65   0.00           0.64   
2    Argentina  ARG  2002    0.07          35.49   0.00           0.92   
3    Argentina  ARG  2003    0.08          33.47   0.00           1.06   
4    Argentina  ARG  2004    0.07          30.14   0.00           0.92   
..         ...  ...   ...     ...            ...    ...            ...   
274  Venezuela  VEN  2016    0.09          62.33   0.01           0.00   
275  Venezuela  VEN  2017    0.09          59.90   0.01           0.00   
276  Venezuela  VEN  2018    0.09          57.57   0.01           0.00   
277  Venezuela  VEN  2019    0.09          72.00   0.01           0.00   
278  Venezuela  VEN  2020    0.09          72.00   0.01           0.00   

       region      sub-region  
0    Americas  America do Sul  
1    Americas  America do Sul  
2    Americas  America do Sul  
3    Americas  America do Sul  
4    Americas  America do Sul  
..        ...             ...  
274  Americas  America do Sul  
275  Americas  America do Sul  
276  Americas  America do Sul  
277  Americas  America do Sul  
278  Americas  America do Sul  

[279 rows x 9 columns]

In [34]:
df.describe()

Ano      Eolica  Hidroeletrica       Solar  Outras Fontes
count   279.000000  279.000000     279.000000  279.000000     279.000000
mean   2010.236559    1.673584      48.945663    0.305663       2.894265
std       6.209875    7.898223      94.612778    1.437319       9.329809
min    2000.000000    0.000000       0.000000    0.000000       0.000000
25%    2005.000000    0.000000       2.150000    0.000000       0.000000
50%    2010.000000    0.000000      19.980000    0.000000       0.170000
75%    2016.000000    0.080000      47.560000    0.010000       1.240000
max    2021.000000   72.240000     424.050000   12.480000      52.390000

In [35]:
df.skew()

Ano              0.005663
Eolica           6.574776
Hidroeletrica    2.934351
Solar            6.265284
Outras Fontes    4.268419
dtype: float64

In [36]:
#IQR
Q1 = df.quantile(0.25)
Q3 = df.quantile(0.75)
IQR = Q3 - Q1
IQR

Ano              11.00
Eolica            0.08
Hidroeletrica    45.41
Solar             0.01
Outras Fontes     1.24
dtype: float64

In [37]:
df['Total_Energia_Renovavel'] = df[['Eolica','Hidroeletrica','Solar','Outras Fontes']].sum(axis=1)

In [38]:
df

Pais Code   Ano  Eolica  Hidroeletrica  Solar  Outras Fontes  \
0    Argentina  ARG  2000    0.04          28.49   0.00           0.70   
1    Argentina  ARG  2001    0.05          36.65   0.00           0.64   
2    Argentina  ARG  2002    0.07          35.49   0.00           0.92   
3    Argentina  ARG  2003    0.08          33.47   0.00           1.06   
4    Argentina  ARG  2004    0.07          30.14   0.00           0.92   
..         ...  ...   ...     ...            ...    ...            ...   
274  Venezuela  VEN  2016    0.09          62.33   0.01           0.00   
275  Venezuela  VEN  2017    0.09          59.90   0.01           0.00   
276  Venezuela  VEN  2018    0.09          57.57   0.01           0.00   
277  Venezuela  VEN  2019    0.09          72.00   0.01           0.00   
278  Venezuela  VEN  2020    0.09          72.00   0.01           0.00   

       region      sub-region  Total_Energia_Renovavel  
0    Americas  America do Sul                    29.23  
1    Americas  America do Sul                    37.34  
2    Americas  America do Sul                    36.48  
3    Americas  America do Sul                    34.61  
4    Americas  America do Sul                    31.13  
..        ...             ...                      ...  
274  Americas  America do Sul                    62.43  
275  Americas  America do Sul                    60.00  
276  Americas  America do Sul                    57.67  
277  Americas  America do Sul                    72.10  
278  Americas  America do Sul                    72.10  

[279 rows x 10 columns]

## Produção de Energia por Ano

In [39]:
fig = px.histogram(df, x="Ano", y="Eolica",text_auto='.2s',

                   title = 'Soma da produção de energia eolica em TW por Ano',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)

fig.show()

In [40]:
fig = px.histogram(df, x="Ano", y="Hidroeletrica",text_auto='.2s',

                   title = 'Soma da produção de energia hidroeletrica em TW por Ano',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

In [41]:
fig = px.histogram(df, x="Ano", y="Solar",text_auto='.2s',

                   title = 'Soma da produção de energia solar por Ano',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [42]:
fig = px.histogram(df, x="Ano", y="Outras Fontes",text_auto='.2s',

                   title = 'Soma da produção de energia por outas fontes em TW por Ano',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

In [43]:
fig = px.histogram(df, x="Ano", y="Total_Energia_Renovavel",text_auto='.2s',

                   title = 'Soma do total da produção de energia renovavel em TW por Ano',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

## Produção de energia por país

    O Brasil representa, apoximadamente 0.60 do percentual de energia renovavel produzida na America Do Sul, é importante resaltar o quanto o Brasil é privilegiado por condições naturais  que contribuem para isso e um amplo território, além de uma maior população, assim um maior consumo e para suprir isso é necessario uma maior produção.
    
    No caso do Chile houve um grande crescimento na produção de energia solar, isso se deve ao Chile concentra a maior central de energia solar da América do Sul, mais precisamente no Deserto de Atacama, esse aumento se da a partir de 2014, ocorrendo um aumento de 514.5%, entre os anos de 2015 e 2020

In [44]:
fig = px.histogram(df, x="Pais", y="Eolica",text_auto='.2s',

                   title = 'Soma da produção de energia eolica em TW por País',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

In [45]:
fig = px.histogram(df, x="Pais", y="Hidroeletrica",text_auto='.2s',

                   title = 'Soma da produção de energia hidroeletrica em TW por País',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

In [46]:
fig = px.histogram(df, x="Pais", y="Solar",text_auto='.2s',

                   title = 'Soma da produção de energia solar por Ano',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [47]:
fig = px.histogram(df, x="Pais", y="Outras Fontes",text_auto='.2s',

                   title = 'Soma da produção de energia renovavel por outas fontes por Pais em TWh',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [48]:
fig = px.histogram(df, x="Pais", y="Total_Energia_Renovavel",text_auto='.2s',

                   title = 'Soma do total da produção de energia renovavel em TW por Pais',

             height=400)
fig.update_layout(bargap=0.2)
fig.show()

In [49]:
total_ener = df['Total_Energia_Renovavel'].sum()
total_br = df[df['Pais'] == 'Brazil']['Total_Energia_Renovavel'].sum()

In [50]:

print(f'total : {total_ener}')

print(f'total br : {total_br}')

total : 15015.550000000001
total br : 8987.279999999999


In [51]:
perct_br = (total_br/total_ener)
perct_br

0.5985315223218596

In [52]:

solar_Chile_2015 = df[(df['Pais'] == 'Chile')&(df['Ano'] == 2015)]['Solar'].sum()

solar_Chile_2020 = df[(df['Pais'] == 'Chile')&(df['Ano'] == 2020)]['Solar'].sum()

In [53]:
print(f'2015 {solar_Chile_2015}')
print(f'202 {solar_Chile_2020}')

2015 1.24
202 7.62


In [54]:
((solar_Chile_2020-solar_Chile_2015)/solar_Chile_2015)

5.145161290322581

In [55]:
fig = px.line(df, x="Ano", y="Eolica", color='Pais')
fig.show()

In [56]:
fig = px.line(df, x="Ano", y="Hidroeletrica", color='Pais')
fig.show()

In [57]:
fig = px.line(df, x="Ano", y="Solar", color='Pais')
fig.show()

In [58]:
fig = px.line(df, x="Ano", y="Outras Fontes", color='Pais')
fig.show()

In [59]:

fig = px.line(df, x="Ano", y="Total_Energia_Renovavel", color='Pais')
fig.show()

## Brasil

    Durante o periodo que vai de 2000 a 2021 a energia proveniente de hidroeletricas em relação a produção total de energia renovavel no pais foi de 0.88.  No ano de 2002, o governo brasileiro criou o Programa de Incentivo às Fontes Alternativas de Energia Elétrica (Proinfa), cujo objetivo era ampliar as matrizes que geram menos impactos ambientais, dentre elas a eólica, a solar, a geotérmica e outras, nesse período houve um crescimento da produção de energia eólica, entre 2010 e 2020 a produção de energia eolica passou de 2.18 TWh para 57.05 TWh, motivada pela produção de energia eólica nos estados da Região Nordeste do país, o crescimento continua entre 2020 e 2021 houve um aumento de 26.64% na produção.
    
    Já a energia solar teve produção mais significativa a partir de 2015, e foi crescento conforme os anos, existido um salto significativo apos 2018, quadruplicando a produção entre 2018 e 2021, passando de 3.46 TWh para 12.48 Twh. No período também é perceptível o aumento da produção por outras fontes como biomassa;
    

In [60]:
df_BR= df[df['Pais'] == 'Brazil']

In [61]:
df_BR

Pais Code   Ano  Eolica  Hidroeletrica  Solar  Outras Fontes    region  \
44  Brazil  BRA  2000    0.00         301.36   0.00           7.41  Americas   
45  Brazil  BRA  2001    0.04         265.20   0.00           8.47  Americas   
46  Brazil  BRA  2002    0.06         283.23   0.00           9.66  Americas   
47  Brazil  BRA  2003    0.06         302.56   0.00          11.26  Americas   
48  Brazil  BRA  2004    0.06         317.59   0.00          11.78  Americas   
49  Brazil  BRA  2005    0.09         334.08   0.00          12.79  Americas   
50  Brazil  BRA  2006    0.24         345.32   0.00          13.99  Americas   
51  Brazil  BRA  2007    0.64         370.27   0.00          16.97  Americas   
52  Brazil  BRA  2008    0.84         365.86   0.00          18.91  Americas   
53  Brazil  BRA  2009    1.24         387.08   0.00          21.81  Americas   
54  Brazil  BRA  2010    2.18         399.26   0.00          30.52  Americas   
55  Brazil  BRA  2011    2.70         424.05   0.00          31.29  Americas   
56  Brazil  BRA  2012    5.05         411.19   0.01          34.35  Americas   
57  Brazil  BRA  2013    6.58         387.08   0.00          39.22  Americas   
58  Brazil  BRA  2014   12.21         369.70   0.02          45.05  Americas   
59  Brazil  BRA  2015   21.63         356.15   0.06          47.70  Americas   
60  Brazil  BRA  2016   33.49         377.10   0.09          49.18  Americas   
61  Brazil  BRA  2017   42.37         367.20   0.83          50.51  Americas   
62  Brazil  BRA  2018   48.47         385.08   3.46          51.97  Americas   
63  Brazil  BRA  2019   55.99         393.90   6.66          52.39  Americas   
64  Brazil  BRA  2020   57.05         409.50  10.75          50.92  Americas   
65  Brazil  BRA  2021   72.24         371.81  12.48          48.97  Americas   

        sub-region  Total_Energia_Renovavel  
44  America do Sul                   308.77  
45  America do Sul                   273.71  
46  America do Sul                   292.95  
47  America do Sul                   313.88  
48  America do Sul                   329.43  
49  America do Sul                   346.96  
50  America do Sul                   359.55  
51  America do Sul                   387.88  
52  America do Sul                   385.61  
53  America do Sul                   410.13  
54  America do Sul                   431.96  
55  America do Sul                   458.04  
56  America do Sul                   450.60  
57  America do Sul                   432.88  
58  America do Sul                   426.98  
59  America do Sul                   425.54  
60  America do Sul                   459.86  
61  America do Sul                   460.91  
62  America do Sul                   488.98  
63  America do Sul                   508.94  
64  America do Sul                   528.22  
65  America do Sul                   505.50

In [62]:
sum_hidro = df_BR['Hidroeletrica'].sum()
sum_total = df_BR['Total_Energia_Renovavel'].sum()

In [63]:
porct_hidro = sum_hidro/sum_total


In [64]:
porct_hidro


0.8817539900837629

In [65]:
df_BR = df_BR.drop(['Pais','Code'],axis=1)

In [66]:
df_BR.describe()

Ano     Eolica  Hidroeletrica      Solar  Outras Fontes  \
count    22.000000  22.000000      22.000000  22.000000      22.000000   
mean   2010.500000  16.510455     360.207727   1.561818      30.232727   
std       6.493587  23.480244      43.092050   3.616609      17.179125   
min    2000.000000   0.000000     265.200000   0.000000       7.410000   
25%    2005.250000   0.127500     336.890000   0.000000      13.090000   
50%    2010.500000   2.440000     369.985000   0.000000      30.905000   
75%    2015.750000  30.525000     387.080000   0.082500      48.652500   
max    2021.000000  72.240000     424.050000  12.480000      52.390000   

       Total_Energia_Renovavel  
count                22.000000  
mean                408.512727  
std                  74.162403  
min                 273.710000  
25%                 350.107500  
50%                 426.260000  
75%                 459.405000  
max                 528.220000

In [67]:
fig = px.bar(df_BR, x="Ano", y="Eolica",text_auto='.2s',

                   title = 'Produção de energia Eolica por Ano no Brasil',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [68]:
ano_eolica_2010 = df_BR[df_BR['Ano'] == 2010]['Eolica'].sum()
print(f'Ano 2010 : {ano_eolica_2010}')
ano_eolica_2020 = df_BR[df_BR['Ano'] == 2020]['Eolica'].sum()
print(f'Ano 2020 : {ano_eolica_2020}')

Ano 2010 : 2.18
Ano 2020 : 57.05


In [69]:
ano_eolica_2021 = df_BR[df_BR['Ano'] == 2021]['Eolica'].sum()
print(f'Ano 2021 : {ano_eolica_2021}')

Ano 2021 : 72.24


In [70]:
print((ano_eolica_2021 - ano_eolica_2020)/ano_eolica_2020)

0.2662576687116564


In [71]:
fig = px.bar(df_BR, x="Ano", y="Hidroeletrica",text_auto='.2s',

                   title = 'Produção de energia Hidroeletrica por Ano no Brasil',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [72]:
fig = px.bar(df_BR, x="Ano", y="Solar",text_auto='.2s',

                   title = 'Produção de energia Solar por Ano no Brasil',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [73]:
ano_solar_2018 = df_BR[df_BR['Ano'] == 2018]['Solar'].sum()
print(f'Ano 2018 : {ano_solar_2018}')
ano_solar_2021 = df_BR[df_BR['Ano'] == 2021]['Solar'].sum()
print(f'Ano 2021 : {ano_solar_2021}')

Ano 2018 : 3.46
Ano 2021 : 12.48


In [74]:
sum_solar = df_BR['Solar'].sum()
sum_total = df_BR['Total_Energia_Renovavel'].sum()


In [75]:
fig = px.bar(df_BR, x="Ano", y="Outras Fontes",text_auto='.2s',

                   title = 'Produção de energia renovavel por outras fontes por Ano no Brasil em TWh',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [76]:
fig = px.bar(df_BR, x="Ano", y="Total_Energia_Renovavel",text_auto='.2s',

                   title = 'Produção total de energia renovavel por Ano no Brasil em TWh',

             height=400)
fig.update_layout(bargap=0.2)
fig.update_traces(textfont_size=12, textangle=0, textposition="outside", cliponaxis=False)
fig.show()

In [77]:
data_BR = pd.melt(df_BR, id_vars=['Ano'], value_vars=['Eolica','Hidroeletrica','Solar','Outras Fontes','Total_Energia_Renovavel'], var_name='Variável', value_name='Valor')

In [78]:
data_BR

Ano                 Variável   Valor
0    2000                   Eolica    0.00
1    2001                   Eolica    0.04
2    2002                   Eolica    0.06
3    2003                   Eolica    0.06
4    2004                   Eolica    0.06
..    ...                      ...     ...
105  2017  Total_Energia_Renovavel  460.91
106  2018  Total_Energia_Renovavel  488.98
107  2019  Total_Energia_Renovavel  508.94
108  2020  Total_Energia_Renovavel  528.22
109  2021  Total_Energia_Renovavel  505.50

[110 rows x 3 columns]

In [79]:
data_BR.groupby('Variável').describe()

Ano                                             \
                        count    mean       std     min      25%     50%   
Variável                                                                   
Eolica                   22.0  2010.5  6.493587  2000.0  2005.25  2010.5   
Hidroeletrica            22.0  2010.5  6.493587  2000.0  2005.25  2010.5   
Outras Fontes            22.0  2010.5  6.493587  2000.0  2005.25  2010.5   
Solar                    22.0  2010.5  6.493587  2000.0  2005.25  2010.5   
Total_Energia_Renovavel  22.0  2010.5  6.493587  2000.0  2005.25  2010.5   

                                         Valor                                 \
                             75%     max count        mean        std     min   
Variável                                                                        
Eolica                   2015.75  2021.0  22.0   16.510455  23.480244    0.00   
Hidroeletrica            2015.75  2021.0  22.0  360.207727  43.092050  265.20   
Outras Fontes            2015.75  2021.0  22.0   30.232727  17.179125    7.41   
Solar                    2015.75  2021.0  22.0    1.561818   3.616609    0.00   
Total_Energia_Renovavel  2015.75  2021.0  22.0  408.512727  74.162403  273.71   

                                                              
                              25%      50%       75%     max  
Variável                                                      
Eolica                     0.1275    2.440   30.5250   72.24  
Hidroeletrica            336.8900  369.985  387.0800  424.05  
Outras Fontes             13.0900   30.905   48.6525   52.39  
Solar                      0.0000    0.000    0.0825   12.48  
Total_Energia_Renovavel  350.1075  426.260  459.4050  528.22

In [80]:
fig = px.line(data_BR, x="Ano", y="Valor", color='Variável')
fig.show()